crates/sui-indexer-alt-jsonrpc/src/api/transactions/filter.rs

```diff
@@ -394,11 +397,28 @@ async fn from_sequence_numbers(
         .transpose()
         .context("Failed to encode next cursor")?;
 
-    let digests = ctx
-        .pg_loader()
-        .load_many(rows.iter().map(|&seq| TxDigestKey(seq as u64)))
-        .await
-        .context("Failed to load transaction digests")?;
+    // Get digests from the table and retry if any of the digests are not found.
+    let retries = ctx.config().transactions.tx_digest_retry_count;
+    let retry_interval =
+        Duration::from_millis(ctx.config().transactions.tx_digest_retry_interval_ms);
+    let mut digests: HashMap<TxDigestKey, StoredTxDigest> = HashMap::new();
+    for _ in 0..retries {
+        digests = ctx
+            .pg_loader()
+            .load_many(rows.iter().map(|&seq| TxDigestKey(seq as u64)))
+            .await
+            .context("Failed to load transaction digests")?;
+
+        // Only retry if the number of digests is less than the number of rows.
+        if digests.len() == rows.len() {
```

Comment:
Is it possible for the number of results returned to be greater than the number of rows requested? Should we add an `ensure` and error out if that happens?

Reply:
No, it's not possible (but also the suggestion above avoids this issue).